encord_agents/core/video.py

```diff
@@ -67,3 +67,32 @@ def iter_video(video_path: Path) -> Iterator[Frame]:
         frame_num += 1
 
     cap.release()
+
+
+def iter_video_with_indices(video_path: Path, frame_indices: Iterable[int]) -> Iterator[Frame]:
+    """
+    Iterate video frame by frame with specified frame indices.
+
+    Args:
+        video_path: The file path to the video you wish to iterate.
+        frame_indices: The frame indices to iterate over.
+
+    Yields:
+        Frames from the video.
+
+    """
```

Comment:
I would add the "side-effect" of sorting the explicit frames such that we avoid seeking in both directions. That should (in some cases but not all) result in higher speed.